### PR TITLE
TST: Split/parameterize indexing tests

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -54,42 +54,45 @@ def test_loc_getitem_label(self):
...
+            ["f", ["floats"], None],
+            [20, ["ints", "uints", "mixed"], None],
+            [20, ["labels"], None],
+            [20, ["ts"], 0],
+            [20, ["floats"], 0],
+        ],
+    )
+    def test_loc_getitem_label_out_of_range(self, key, typs, axes):
 
         # out of range label
```
Comment (jbrockmendel) [inline pandas/tests/indexing/test_loc.py:70]: does this comment apply to all of these or just the one currently on L60?

File: pandas/tests/indexing/test_scalar.py [hunk 2]
```diff
@@ -103,24 +76,25 @@ def test_at_iat_coercion(self):
         xp = s.values[5]
         assert result == xp
 
+    @pytest.mark.parametrize("idx", [lambda ser: ser.iat[1], lambda ser: ser.iloc[1]])
```
Comment (jbrockmendel) [inline pandas/tests/indexing/test_scalar.py:79]: there's a fixture indexer_al for at/loc, could make a similar one for iat/iloc

Comment (mroeschke) [reply pandas/tests/indexing/test_scalar.py]: Sure. Made a fixture

File: pandas/tests/indexing/test_loc.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/indexing/test_loc.py:70]: I think this comment applies to all cases.

**Pull request merged**